### DISCUSSION TITLE
Will we ever have other physical constraints besides the copies of the Godot ones?

### DISCUSSION THREAD

**Discussion by towerscripter1386** [godot-jolt/godot-jolt#999] [Questions] [answered]
Such as Gear constraint, Pulley constraint, Rack And Pinion constraint and etc

**Comment by mihe** [comment, marked as answer]
In this extension, no. Once this has been migrated into Godot itself as an actual engine module then maybe. There still isn't a clear plan for how to deal with some of these more Jolt-specific features though.

**Comment by addmix** [comment]
Nearing the release of 4.4, which includes Jolt, has there been any more thought towards these constraint types? I think these are very valuable constraints, and it's a shame that we don't have them (yet?)

**Reply by mihe** [reply]
> has there been any more thought towards these constraint types?

Not really, no. There is a brief proposal made by another community member that you can upvote though: godotengine/godot-proposals#11338

> I think these are very valuable constraints, and it's a shame that we don't have them (yet?)

I do think Jolt's `FixedConstraint` and `DistanceConstraint` makes sense to add to the shared interface at some point, since they're likely widely applicable to a lot of other physics engines out there, but the other ones are a tougher sell. It's unclear how we'd best go about exposing those.

**Reply by addmix** [reply]
I figure if the Godot foundation is content with the 6DOF joint types being unimplemented, it shouldn't be an issue for there to be other joint/constraint types that aren't implemented in all physics engine implementations. Here's to hoping someone can figure it out eventually.
